tests/test_unit/test_cdf_tk/test_commands/test_migrate.py

```diff
@@ -70,17 +67,61 @@ def verify_capabilities(desired_capabilities: ComparableCapability) -> list[Capa
 
             client.iam.verify_capabilities = verify_capabilities
             with pytest.raises(AuthenticationError) as exc_info:
-                cmd.validate_access(client, ["my_instance_space"], [123])
+                cmd.validate_access(client, ["my_instance_space"], ["dummy_space"], [123])
         error = exc_info.value
         assert isinstance(error, AuthenticationError)
         assert "Missing required capabilities" in error.args[0]
         assert error.args[1] == expected_missing
 
-    def test_validate_instance_source_exists_raise(self) -> None:
+    def test_validate_migration_model_available(self) -> None:
         with monkeypatch_toolkit_client() as client:
-            client.data_modeling.views.retrieve.return_value = ViewList([])
+            client.data_modeling.data_models.retrieve.return_value = DataModelList([])
             with pytest.raises(ToolkitMigrationError):
-                BaseMigrateCommand.validate_instance_source_exists(client)
+                BaseMigrateCommand.validate_migration_model_available(client)
+
+    def test_validate_migration_model_available_multiple_models(self) -> None:
+        """Test that multiple models raises an error."""
+        with monkeypatch_toolkit_client() as client:
+            # Create mock models with the expected MODEL_ID
+            model1 = MagicMock(spec=DataModel)
+            model1.as_id.return_value = MODEL_ID
+            model2 = MagicMock(spec=DataModel)
+            model2.as_id.return_value = MODEL_ID
+
+            client.data_modeling.data_models.retrieve.return_value = DataModelList([model1, model2])
+
+            with pytest.raises(ToolkitMigrationError) as exc_info:
+                BaseMigrateCommand.validate_migration_model_available(client)
+
+            assert "Multiple migration models" in str(exc_info.value)
+
+    def test_validate_migration_model_available_missing_views(self) -> None:
+        """Test that a model with missing views raises an error."""
+        with monkeypatch_toolkit_client() as client:
+            model = MagicMock(spec=DataModel)
+            model.as_id.return_value = MODEL_ID
+            # Model has views but missing the required ones
+            model.views = [INSTANCE_SOURCE_VIEW_ID]  # Missing VIEW_SOURCE_VIEW_ID
+
+            client.data_modeling.data_models.retrieve.return_value = DataModelList([model])
+
+            with pytest.raises(ToolkitMigrationError) as exc_info:
+                BaseMigrateCommand.validate_migration_model_available(client)
+
+            assert "Invalid migration model. Missing views" in str(exc_info.value)
```

Comment:
```suggestion
            with pytest.raises(ToolkitMigrationError, match="Invalid migration model. Missing views"):
                BaseMigrateCommand.validate_migration_model_available(client)
```
this can be done in one line